toolkit/primitives/domain/src/lib.rs

```diff
@@ -476,6 +476,33 @@ impl UtxoInfo {
 	}
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(rename_all = "lowercase"))]
+pub enum NetworkType {
+	Mainnet,
+	Testnet,
+}
+
+#[cfg(feature = "std")]
+impl NetworkType {
+	pub fn as_string(&self) -> String {
```

Comment:
Is it required?
You define `Display`, so I think we have `.to_string()` for free, don't we?